app/actions/package_update.rb

```diff
@@ -9,6 +9,11 @@ def initialize
 
     def update(package, message)
       package.db.transaction do
+        if package.type == PackageModel::DOCKER_TYPE
+          package.docker_username = message.username unless message.username.nil?
```

Comment:
I decided for .nil? because you can push a docker app like this:
`CF_DOCKER_PASSWORD='' cf push my-test-app -o cloudfoundry/test-app --docker-username ''`
so I thought I be consistent and allow empty string. But if I change it to .present? and you do a package update with empty string as username or password, the package doesn't get updated. What would be the best solution?

Reply:
Ok, if you update an app with an empty name, you get an error `{"errors":[{"detail":"name presence, name format","title":"CF-UnprocessableEntity","code":10008}]}`. And if you try to update with empty lifecycle or type then there are also errors thrown `{"errors":[{"detail":"Lifecycle data must be an object","title":"CF-UnprocessableEntity","code":10008}]}`,
`{"errors":[{"detail":"Lifecycle type is not included in the list: buildpack, docker","title":"CF-UnprocessableEntity","code":10008}]}`. Maybe this would also be an option if we decide for not allowing empty strings to improve the error message.

Reply:
Okay, if you can create a package with empty strings for username and password, that should also be supported for the update scenario. So `nil?` seems to be the correct check.